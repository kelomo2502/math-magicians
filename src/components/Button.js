import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styled from 'styled-components';

const Btn = styled.button`
  border: none;
  background: ${(props) => (props.primary ? '#e0e0e0' : '#f5913e')};
  display: flex;
  justify-content: center;
  align-items: center;
`;

export default class Button extends Component {
  constructor(props) {
    super(props);
    const { typeBtn, btnClass, primary } = this.props;
    this.state = {
      nameBtn: typeBtn,
      btnClass,
      primary,
    };
    this.btnOnClick = this.btnOnClick.bind(this);
  }

  btnOnClick(nameButton) {
    const { btnOnClick } = this.props;
    btnOnClick(nameButton);
  }

  render() {
    const { nameBtn, btnClass, primary } = this.state;
    return <Btn primary={primary} type="button" className={btnClass} onClick={() => this.btnOnClick(nameBtn)}>{ nameBtn }</Btn>;
  }
}

Button.propTypes = {
  typeBtn: PropTypes.string.isRequired,
  btnClass: PropTypes.string,
  btnOnClick: PropTypes.func,
  primary: PropTypes.bool,
};

Button.defaultProps = {
  btnClass: 'btn-light',
  primary: false,
  btnOnClick: () => {},
};
