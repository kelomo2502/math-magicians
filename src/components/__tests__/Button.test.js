import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Button
        typeBtn="AC"
        btnOnClick={() => {}}
        primary
        btnClass="btn-clear"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
