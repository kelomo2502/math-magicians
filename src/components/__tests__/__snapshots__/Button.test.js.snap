// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="sc-bczRLJ kLLeMr btn-clear"
  onClick={[Function]}
  type="button"
>
  AC
</button>
`;
